server/server_test.go

```diff
@@ -247,3 +248,43 @@ func TestCacheIndexHandler(t *testing.T) {
 		t.Errorf("want: 200; got: %d.\n\tcan't delete keys.", resp.StatusCode)
 	}
 }
+
+func TestInvalidPutWhenExceedShardCap(t *testing.T) {
+	t.Parallel()
+	req := httptest.NewRequest("PUT", testBaseString+"/api/v1/cache/putKey", bytes.NewBuffer(blob('a', 8*1024*1024)))
+	rr := httptest.NewRecorder()
+
+	putCacheHandler(rr, req)
+	resp := rr.Result()
+
+	if resp.StatusCode != 500 {
+		t.Errorf("want: 500; got: %d", resp.StatusCode)
+	}
+}
+
+func TestInvalidPutWhenReading(t *testing.T) {
+	t.Parallel()
+	req := httptest.NewRequest("PUT", testBaseString+"/api/v1/cache/putKey", errReader(0))
+	rr := httptest.NewRecorder()
+
+	putCacheHandler(rr, req)
+	resp := rr.Result()
+
+	if resp.StatusCode != 500 {
+		t.Errorf("want: 500; got: %d", resp.StatusCode)
+	}
+}
+
+func blob(char byte, len int) []byte {
```

Comment:
https://godoc.org/bytes#Repeat

Reply:
awesome , I think all blob function should be replaced to bytes.Repeat.